Automattic/jetpack
# Social | Fix no connections UI for editor

Right now, when there are no connections, the UI in the editor points to WPCOM connections page to add an account


![Image](https://github.com/Automattic/jetpack-reach/assets/18226415/101a8ca9-db25-4690-b460-e72bd30f01fa)

This PR updates that UI based on the featured flag to render the connections modal instead of that external link.

## Proposed changes:
* I took this opportunity to rename `AddConnnectionModal` to `ManageConnectionsModal`
* Extract modal state management to re-usable component named `ManageConnectionsModalWithTrigger`. The benefit of this change is that the modal is used at many places and we need to maintain the state of the modal and the corresponding keyring result at all these places. Thus, I extracted that part of the logic to the above mentioned component, so that the consumer component only renders the modal with the trigger button UI. This also fixes the problem of having to use `ThemeProvider` everywhere, because now the `ThemeProvider` is rendered inside the above component.
* Wire up connections modal in editor when there are no connections

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






- Ensure that feature flag is ON - `define( 'JETPACK_SOCIAL_USE_ADMIN_UI_V1', true );`
- Test the following for both Jetpack and Social plugin
    - Confirm that everything works as before
    - Remove all connections
    - Turn OFF feature flag
    - Goto post editor
    - Confirm that under "Share this post" section it renders the old UI - a link to WPCOM connections page.
    - Turn ON the feature flag and reload the editor page
    - Confirm that now you see "Connect an account" button
    - Click on it
    - Confirm that it opens the modal and not the external link to WPCOM
    - Add a connection here
    - Confirm that it works as expected
    - Confirm that now the + button behaves same as before.

<table>
    <tbody>
        <tr>
            <th>BEFORE</th>
            <th>AFTER</th>
        </tr>
        <tr>
            <td>
<img width="283" alt="Screenshot 2024-05-27 at 4 03 22 PM" src="https://github.com/Automattic/jetpack/assets/18226415/b35973fe-718c-4cdf-9dab-53e62d4f43c6">

</td>
            <td><img width="285" alt="Screenshot 2024-05-27 at 4 02 50 PM" src="https://github.com/Automattic/jetpack/assets/18226415/7f9387e4-998e-4833-b5a1-e5fe4765a9e7"></td>
        </tr>
    </tbody>
</table>
